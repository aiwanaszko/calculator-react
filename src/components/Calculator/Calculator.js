import React from "react";
import Button from "./components/Button";

export default function Calculator() {
  return (
    <div className="container">
      <div className="row">
        <Button>7</Button>
        <Button>8</Button>
        <Button>9</Button>
        <Button>+</Button>
      </div>
      <div className="row">
        <Button>4</Button>
        <Button>5</Button>
        <Button>6</Button>
        <Button>-</Button>
      </div>
      <div className="row">
        <Button>1</Button>
        <Button>2</Button>
        <Button>3</Button>
        <Button>-</Button>
      </div>
      <div className="row">
        <Button>C</Button>
        <Button>0</Button>
        <Button>.</Button>
        <Button>=</Button>
      </div>
    </div>
  );
}
