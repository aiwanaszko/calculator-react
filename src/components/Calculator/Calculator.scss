@import url("https://fonts.googleapis.com/css2?family=Montserrat:wght@200;300&display=swap");

.calculator-wrapper {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
  width: 100%;
  background: rgb(131, 58, 180);
  background: linear-gradient(
    90deg,
    rgba(131, 58, 180, 1) 0%,
    rgba(234, 72, 93, 1) 50%,
    rgba(217, 181, 44, 1) 100%
  );
  font-family: "Montserrat", sans-serif;
  font-weight: bold;
}

.container {
  background-color: black;
  border-radius: 5px;
  width: 400px;
}

.row {
  display: flex;
  justify-content: space-between;
  padding: 5px;
}

.logo {
  width: 120px;
  height: auto;
  position: absolute;
  top: 20px;
  left: 20px;
}
