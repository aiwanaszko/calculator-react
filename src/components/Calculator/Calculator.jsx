import React, { Component } from "react";
import Display from "../Display/Display";
import Button from "../Button/Button";
import "./Calculator.scss";
//import logo from "../../public/DCMN_logo.png";

class Calculator extends Component {
  constructor(props) {
    super(props);
    this.state = {
      equation: "",
      result: "",
    };
  }

  addToInput = (val) => {
    this.setState({ equation: this.state.equation + val });
  };

  handleEqual = () => {
    // evaluate
  };

  render() {
    return (
      <div className="calculator-wrapper">
        {/* <img className="logo" src={logo}></img> */}
        <div className="container">
          <Display equation={this.state.equation} result={this.state.result} />
          <div className="row">
            <Button handleClick={this.addToInput}>7</Button>
            <Button handleClick={this.addToInput}>8</Button>
            <Button handleClick={this.addToInput}>9</Button>
            <Button handleClick={this.addToInput}>+</Button>
          </div>
          <div className="row">
            <Button handleClick={this.addToInput}>4</Button>
            <Button handleClick={this.addToInput}>5</Button>
            <Button handleClick={this.addToInput}>6</Button>
            <Button handleClick={this.addToInput}>-</Button>
          </div>
          <div className="row">
            <Button handleClick={this.addToInput}>1</Button>
            <Button handleClick={this.addToInput}>2</Button>
            <Button handleClick={this.addToInput}>3</Button>
            <Button handleClick={this.addToInput}>-</Button>
          </div>
          <div className="row">
            <Button
              handleClick={() => this.setState({ equation: "", result: "" })}
            >
              {this.state.equation === 0 ? "AC" : "C"}
            </Button>
            <Button handleClick={this.addToInput}>0</Button>
            <Button handleClick={this.addToInput}>.</Button>
            <Button>=</Button>
          </div>
        </div>
      </div>
    );
  }
}

export default Calculator;
