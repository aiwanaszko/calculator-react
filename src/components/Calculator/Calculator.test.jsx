import React from "react";
import renderer from "react-test-renderer";
import Calculator from "./Calculator";

it("renders the Calculator correctly", () => {
  const component = renderer.create(<Calculator />).toJSON();
  expect(component).toMatchSnapshot();
});
