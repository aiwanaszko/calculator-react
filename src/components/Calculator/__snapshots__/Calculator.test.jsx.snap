// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Calculator correctly 1`] = `
<div
  className="calculator-wrapper"
>
  <img
    className="logo"
    src="DCMN_logo.png"
  />
  <div
    className="calculator-container"
  >
    <div
      className="display"
    >
      <div
        className="equation-row"
      >
        <input
          readOnly={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="result-row"
      >
        <input
          readOnly={true}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="buttons-container"
    >
      <div
        className="column"
      >
        <div
          className="button-wrapper "
          onClick={[Function]}
        >
          7
        </div>
        <div
          className="button-wrapper "
          onClick={[Function]}
        >
          4
        </div>
        <div
          className="button-wrapper "
          onClick={[Function]}
        >
          1
        </div>
        <div
          className="button-wrapper operator"
          onClick={[Function]}
        >
          AC
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="button-wrapper "
          onClick={[Function]}
        >
          8
        </div>
        <div
          className="button-wrapper "
          onClick={[Function]}
        >
          5
        </div>
        <div
          className="button-wrapper "
          onClick={[Function]}
        >
          2
        </div>
        <div
          className="button-wrapper "
          onClick={[Function]}
        >
          0
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="button-wrapper "
          onClick={[Function]}
        >
          9
        </div>
        <div
          className="button-wrapper "
          onClick={[Function]}
        >
          6
        </div>
        <div
          className="button-wrapper "
          onClick={[Function]}
        >
          3
        </div>
        <div
          className="button-wrapper "
          onClick={[Function]}
        >
          .
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="button-wrapper operator"
          onClick={[Function]}
        >
          +
        </div>
        <div
          className="button-wrapper operator"
          onClick={[Function]}
        >
          -
        </div>
        <div
          className="button-wrapper equality-operator"
          onClick={[Function]}
        >
          =
        </div>
      </div>
    </div>
  </div>
</div>
`;
